### PR TITLE
hal/imxrt: Fix boot from FlexSPI failing

### PR DESCRIPTION
**Pull request by Darchiv**
This change fixes a boot failure when plo image is larger than the hardcoded value in sflashA1Size, but smaller than m_flash size (the linker does not abort). Now (initial) flash size is tied to FlexSPI Configuration Block so that this never happens again. If plo grows too much, the linker will produce an error.

JIRA: RTOS-920

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: imxrt1170-evk.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: ld/armv7m7-imxrt117x.ldt [hunk 1]
```diff
@@ -56,9 +56,12 @@ MEMORY
 	m_dtcm    (rw)  : ORIGIN = 0x20002000 + AREA_KERNEL, LENGTH = FLEXRAM_DTCM_AREA - AREA_KERNEL
 	m_ocram1  (rwx) : ORIGIN = 0x20240000 + AREA_BOOTLOADER, LENGTH = (8 * 32k) - AREA_BOOTLOADER
 	m_ocram2  (rwx) : ORIGIN = 0x202c0000, LENGTH = 512k
-	m_flash   (rx)  : ORIGIN = 0x30000000, LENGTH = 4M
+	m_flash   (rx)  : ORIGIN = 0x30000000, LENGTH = 128k
```
Comment (nalajcie) [inline ld/armv7m7-imxrt117x.ldt:59]: maybe add the same comment as below also here?

File: ld/armv7m7-imxrt117x.ldt [hunk 2]
```diff
@@ -56,9 +56,12 @@ MEMORY
...
 }
 
+/* initial flash size for imxrt boot procedure only */
```
Comment (nalajcie) [inline ld/armv7m7-imxrt117x.ldt:62]: maybe mention BootROM, eg. `initial flash size to  be put into FCB block for imxrt BootROM init procedure only`

**Review by nalajcie — DISMISSED** [review]
I suggest to make the comments more descriptive, other than thad - LGTM

@ziemleszcz do you have any idea why `sflashA1Size` was previously greatly reduced in commit https://github.com/phoenix-rtos/plo/commit/44ccfa5b9ca1aa91b3513e615866a86d806a8ca6

Probably it's just for better portability - but I'm not sure.

File: ld/armv7m7-imxrt117x.ldt (hunk 2 above)
Comment (Darchiv) [reply ld/armv7m7-imxrt117x.ldt]: Done

File: ld/armv7m7-imxrt117x.ldt (hunk 1 above)
Comment (Darchiv) [reply ld/armv7m7-imxrt117x.ldt]: Done

**Pull request merged**